pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/onchistory/OncHistoryUploadService.java

```diff
@@ -105,9 +107,28 @@ public void upload(String fileContent) {
         validateRows(rows);
         log.info("Validated {} rows for onc history upload", rows.size());
 
+        //error out if any exited participants.
```

Comment:
We already check the upload participants in `OncHistoryUploadService.getParticipantIds`. That is the correct place to check for exited as well, avoiding another iteration and additional code.

I suggest changing `ESParticipantIdProvider.getParticipantIdForShortId` to instead call `ElasticSearchService.getParticipantDocumentByShortId`, do the participant ID validation as it does now, and return a struct/record/class that includes the participant ID and `ElasticSearchParticipantDto`, or whatever part of `ElasticSearchParticipantDto` you need.

Does that make sense?

Reply:
OncHistoryUploadService.getParticipantIds does some additional stuff like creating a MR if needed. 
Felt like code might look cluttered even though we save a trip to ES. 
On a second thought since the method is throwing error anyway, I modified the code to check for EXITED status and throw error. Added some documentation mentioning that we are checking for EXITED participants.